packages/playground/website/src/sync.ts

```diff
@@ -0,0 +1,157 @@
+import { phpVars, startPlaygroundWeb } from '@wp-playground/client';
+import { login } from '@wp-playground/blueprints';
+import { FilesystemOperation } from '@php-wasm/universal';
+/** @ts-ignore */
+import bumpAutoIncrements from './bump-auto-increment.php?raw';
+import patchedSqliteTranslator from './class-wp-sqlite-translator.php?raw';
+
+const playground = await startPlaygroundWeb({
+	iframe: document.getElementById('wp') as HTMLIFrameElement,
+	remoteUrl: 'http://localhost:4400/remote.html',
+});
+await login(playground, { username: 'admin', password: 'password' });
+await playground.goTo('/');
+
+await playground.writeFile(
+	'/wordpress/wp-content/plugins/sqlite-database-integration/wp-includes/sqlite/class-wp-sqlite-translator.php',
+	patchedSqliteTranslator
+);
+
+await playground.run({
+	code: bumpAutoIncrements,
+});
+
+class Counter extends Map<string, number> {
+	increment(key: string) {
+		this.set(key, (this.get(key) ?? 0) + 1);
+	}
+	get(key: string): number {
+		return super.get(key) || 0;
+	}
+	decrement(key: string) {
+		const newValue = (this.get(key) ?? 0) - 1;
+		if (newValue > 0) {
+			this.set(key, newValue);
+		} else {
+			this.delete(key);
+		}
+	}
+}
+
+function broadcastChange(scope: string, details: any) {
+	window.top?.postMessage(
+		{
+			type: 'playground-change',
+			scope,
+			details,
+		},
+		'*'
+	);
+}
+
+function onChangeReceived(scope: string, fn: (details: any) => void) {
+	window.addEventListener('message', (event) => {
+		if (event.data.type !== 'playground-change') {
+			return;
+		}
+		if (event.data.scope !== scope) {
+			return;
+		}
+		fn(event.data.details);
+	});
+}
+
+const replayedSqlQueries = new Counter();
+
+playground.onMessage(async (messageString) => {
+	const { type, ...data } = JSON.parse(messageString) as any;
+	if (type !== 'sql') {
+		return;
+	}
+	const firstKeyword = data.query.trim().split(/\s/)[0].toLowerCase();
+	if (firstKeyword === 'select') {
+		return;
+	}
+	if (replayedSqlQueries.get(data.query) > 0) {
+		return;
+	}
+	broadcastChange('sql', data);
+});
+
+onChangeReceived('sql', async (data) => {
+	console.log('[SQL]', data);
+
+	replayedSqlQueries.increment(data.query);
```

Comment:
that's the point of the `Counter`? and incrementing here?

Reply:
Too convoluted to explain 🙈 I simplified everything there and removed the Counter entirely